src/graphs/lca.rs

```diff
@@ -42,7 +42,23 @@ impl LCA {
         let mut tin = vec![0; n];
         let mut p = vec![0; n];
         let mut d = vec![0; n];
-        let order = get_dfs_preorder(adj);
+        let mut order = Vec::with_capacity(n);
+        dfs(
+            adj,
+            |u| {
+                tin[u] = order.len();
+                order.push(u);
+                for &v in &adj[u] {
+                    if v != p[u] {
+                        p[v] = u;
+                        d[v] = d[u] + 1;
+                    }
+                }
+            },
+            |_| {},
+        );
+        //let order = get_dfs_preorder(adj);
+        /*
         for (i, &u) in order.iter().enumerate() {
             tin[u] = i;
```

Comment:
basically just proof of concept; showing it works;

I can clean it up later if we decide to keep it